### PR TITLE
using sub-cache semaphore in subCacheItemNotificationErrorStatus

### PR DESCRIPTION
**Pull request by kzangeli**
Sub-cache semaphore was not taken before updating lastSuccess+lastFailure in the sub cache items ...
Stupid mistake, lots of time wasted ... I apologize :-(

### CODE REVIEW

File: src/lib/cache/subCache.cpp [hunk 1]
```diff
@@ -1353,18 +1367,21 @@ void subCacheItemNotificationErrorStatus(const std::string& tenant, const std::s
     return;
   }
 
+  time_t now = time(NULL);
```
Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: I moved the call to 'time' here to avoid a syscall with the semaphore taken

**Comment by crbrox** [comment]
LGTM

File: src/lib/cache/subCache.cpp [hunk 2]
```diff
@@ -1050,9 +1050,23 @@ int subCacheItemRemove(CachedSubscription* cSubP)
...
+      if (cSubP == subCache.tail)
+      {
+        subCache.tail = prev;
+      }
+
+      // Removing middle item?
+      if (prev != NULL)
+      {
+        prev->next = cSubP->next;
+      }
```
Comment (fgalan) [inline src/lib/cache/subCache.cpp]: LGTM

**Comment by fgalan** [comment]
LGTM

**Pull request merged**